### PR TITLE
Add `useMintContribution` Mutation React-Query

### PR DESCRIPTION
**Pull request by amrro**
## Linear Ticket
[PRO-485 - Split out mint contribution](https://linear.app/govrn/issue/PRO-485/split-out-mint-contribution)

## Description
Refactor mint contribution from context to react-query mutation
___
Related to 
- #225 

Depends on 
- #219

### CODE REVIEW

File: apps/protocol-frontend/src/components/MintModal.tsx [hunk 1]
```diff
@@ -85,11 +89,9 @@ const MintModal = ({ contributions }: MintModalProps) => {
           date_of_submission: original.date_of_submission.toString(),
           engagementDate: original.engagementDate.toString(),
         };
-        mintContribution(originalClean, ipfsContentUri, setMintProgress);
+        mintContribution({ ...originalClean, ipfsContentUri });
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/MintModal.tsx:92]: This is a good opportunity to make an ipfs hook and call it in the mint contribution hook

Comment (amrro) [reply apps/protocol-frontend/src/components/MintModal.tsx:92]: Totally, but in different PR to be used in both mint & bulk mint.

File: apps/protocol-frontend/src/components/MintModal.tsx [hunk 2]
```diff
@@ -36,7 +40,7 @@ const MintModal = ({ contributions }: MintModalProps) => {
     if (minting && mintProgress === mintTotal) {
       setMinting(false);
     }
-  }, [mintProgress]);
+  }, [mintProgress, mintTotal, minting]);
```
[the remark this answers is not in the document]
Comment (alexkeating) [reply apps/protocol-frontend/src/components/MintModal.tsx:43]: Great, which ticket are you referring to?

[the remark this answers is not in the document]
Comment (amrro) [reply apps/protocol-frontend/src/components/MintModal.tsx:43]: https://linear.app/govrn/issue/PRO-517/handle-dismissing-mint-modal-after-finishing

[the remark this answers is not in the document]
Comment (alexkeating) [reply apps/protocol-frontend/src/components/MintModal.tsx]: After some testing I noticed that this pr removed dismissing the modal `setModal({})` clears the modal. So, I added it back and I also removed the useEffect in [this](https://github.com/Govrn-HQ/govrn-monorepo/pull/233/commits/424aa3a0eeb64b4313bdc3be8dbe81408da1306a) commit.

**Review by alexkeating — APPROVED** [review]

File: apps/protocol-frontend/src/components/MintModal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (amrro) [reply apps/protocol-frontend/src/components/MintModal.tsx]: Thanks! 

But Please make sure re-adding `setModal({})` doesn't dismiss the modal immediately if you tap the mint button.

**Pull request merged**